resources/apalache.nix

```diff
@@ -0,0 +1,60 @@
+{
+  apalache-src,
+  pkgs,
+}: let
+  version = "v0.23.1";
+
+  # Patch the build.sbt file so that it does not call the `git describe` command.
+  # This is called by sbt-derivation to resolve the Scala dependencies, however
+  # inside the Nix build environment for sbt-derivation, the git command is
+  # not available, hence the dependency resolution would fail. As a workaround,
+  # we use the version string provided in Nix as the build version.
+  #
+  # Note that the diff has a single space indentation, so auto re-indentation
+  # inside the editor may break the diff.
+  patch = ''
+    diff --git a/build.sbt b/build.sbt
+    index c052ebc8..fa4568d7 100644
+    --- a/build.sbt
+    +++ b/build.sbt
+    @@ -184,7 +184,7 @@
+           // See https://github.com/sbt/sbt-buildinfo
+           buildInfoPackage := "apalache",
+           buildInfoKeys := {
+    -        val build = scala.sys.process.Process("git describe --tags --always").!!.trim
+    +        val build = "${version}"
+             Seq[BuildInfoKey](
+                 BuildInfoKey.map(version) { case (k, v) =>
+                   if (isSnapshot.value) (k -> build) else (k -> v)
+  '';
+in
+  pkgs.sbt.mkDerivation {
+    pname = "apalache";
+    inherit version;
+
+    depsSha256 = "sha256-xKLKlkOHysNtSCDtj9JKwLYyCCuRrc36+QsBWFjLnFI=";
+
+    src = apalache-src;
+
+    patches = [
+      (builtins.toFile "diff.patch" patch)
+    ];
+
+    buildPhase = ''
+      sbt apalacheCurrentPackage
+    '';
+
+    installPhase = ''
+      mkdir -p $out/lib
+      mkdir -p $out/bin
+
+      cp -r target/universal/current-pkg/lib/apalache.jar $out/lib/apalache.jar
+
+      cat > $out/bin/apalache-mc <<- EOM
+      #!${pkgs.bash}/bin/bash
+      exec ${pkgs.jre}/bin/java -Xmx4096m -jar $out/lib/apalache.jar $@
```

Comment:
It looks like the wrapper is not forwarding the command line arguments correctly.

Reply:
kk, should be fixed!

Reply:
Fixed that in [4c8ceb6](https://github.com/informalsystems/cosmos.nix/pull/80/commits/4c8ceb64078f10b5bb03b9fd6646c3fa3f330f17). The `$@` has to be escaped so that it is not replaced by `cat > $out/bin/apalache-mc <<- EOM`.

Reply:
I don't think we need the quotes, I think it can just be `\$@`